internal/k8s/controller.go

```diff
@@ -3302,21 +3302,25 @@ func (lbc *LoadBalancerController) isHealthCheckEnabled(ing *networking.Ingress)
 	return false
 }
 
-// ValidateSecret validates that the secret follows the TLS Secret format.
-// For NGINX Plus, it also checks if the secret follows the JWK Secret format.
-func (lbc *LoadBalancerController) ValidateSecret(secret *api_v1.Secret) error {
-	err1 := ValidateTLSSecret(secret)
-	if !lbc.isNginxPlus {
-		return err1
-	}
-
-	err2 := ValidateJWKSecret(secret)
+// ValidateSecret validates that the secret follows the TLS, MTLS or JWK Secret format.
+func (lbc *LoadBalancerController) ValidateSecret(secret *api_v1.Secret) (err error) {
```

Comment:
this should have unit tests (?)

Reply:
unit test would help! 
I think the function could be simplified to
```
	_, err := GetSecretKind(secret)
	return err
```